### PR TITLE
Alias copy patch

### PR DESCRIPTION
**Pull request by jjsjann123**
fixes the assertion from #1325 on our devel branch.

1. update alias information after graph mutation
2. patch unsqueeze: i. support negative dimension; ii. fixing range check

### CODE REVIEW

**Review by kevinstephano — APPROVED** [review]
Lgtm

File: torch/csrc/jit/codegen/cuda/ops/alias.cpp [hunk 1]
```diff
@@ -94,7 +94,10 @@ TensorView* squeeze(
 }
 
 TensorView* unsqueeze(TensorView* x, int64_t dim) {
-  TORCH_INTERNAL_ASSERT(dim >= 0 && dim < x->nDims());
+  if (dim < 0) {
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/ops/alias.cpp:100]: Would we need to handle negative axes with `squeeze` as well?
https://github.com/csarofeen/pytorch/blob/85eceb82e0b0a39c8d92f869ae1b6be42198833d/torch/csrc/jit/codegen/cuda/ops/alias.cpp#L85-L91

File: torch/csrc/jit/codegen/cuda/graph_fuser.cpp [hunk 2]
```diff
@@ -2042,6 +2042,9 @@ void replaceAliasOpsWithCopy(std::shared_ptr<Graph>& graph, Block* block) {
         graph->insertNode(graph->create(op_mapping[n->kind()], n->inputs(), 1));
     op_copy->output()->setType(n->output(0)->type());
 
+    // adding newly created value into alias_db;
+    alias_db->createValue(op_copy->output());
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/graph_fuser.cpp:2046]: Do we need to remove the value from `alias_db` in `revertAliasCopyOps`?
https://github.com/csarofeen/pytorch/blob/85eceb82e0b0a39c8d92f869ae1b6be42198833d/torch/csrc/jit/codegen/cuda/graph_fuser.cpp#L2056

**Review by rdspring1 — APPROVED** [review]
LGTM.

File: torch/csrc/jit/codegen/cuda/ops/alias.cpp (hunk 1 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/ops/alias.cpp:100]: Ha, we should, let me update this.

File: torch/csrc/jit/codegen/cuda/graph_fuser.cpp (hunk 2 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/graph_fuser.cpp:2046]: I don't think that's necessary. We'll just leave a dangling value in the set and that shouldn't hurt anything?

**Pull request merged**